samza-core/src/main/scala/org/apache/samza/storage/ContainerStorageManager.java

```diff
@@ -273,491 +200,52 @@ public ContainerStorageManager(
-
-    Map<SystemStreamPartition, TaskSideInputHandler> handlers = new HashMap<>();
-
-    if (this.hasSideInputs) {
-      containerModel.getTasks().forEach((taskName, taskModel) -> {
-
-        Map<String, StorageEngine> taskSideInputStores = sideInputStores.get(taskName);
-        Map<String, Set<SystemStreamPartition>> sideInputStoresToSSPs = new HashMap<>();
-        boolean taskHasSideInputs = false;
-        for (String storeName : taskSideInputStores.keySet()) {
-          Set<SystemStreamPartition> storeSSPs = this.taskSideInputStoreSSPs.get(taskName).get(storeName);
-          taskHasSideInputs = taskHasSideInputs || !storeSSPs.isEmpty();
-          sideInputStoresToSSPs.put(storeName, storeSSPs);
-        }
-
-        if (taskHasSideInputs) {
-          CountDownLatch taskCountDownLatch = new CountDownLatch(1);
-          this.sideInputTaskLatches.put(taskName, taskCountDownLatch);
-
-          TaskSideInputHandler taskSideInputHandler = new TaskSideInputHandler(taskName,
-              taskModel.getTaskMode(),
-              loggedStoreBaseDirectory,
-              taskSideInputStores,
-              sideInputStoresToSSPs,
-              taskSideInputProcessors.get(taskName),
-              this.systemAdmins,
-              this.streamMetadataCache,
-              taskCountDownLatch,
-              clock);
-
-          sideInputStoresToSSPs.values().stream().flatMap(Set::stream).forEach(ssp -> {
-            handlers.put(ssp, taskSideInputHandler);
-          });
-
-          LOG.info("Created TaskSideInputHandler for task {}, taskSideInputStores {} and loggedStoreBaseDirectory {}",
-              taskName, taskSideInputStores, loggedStoreBaseDirectory);
-        }
-      });
-    }
-    return handlers;
-  }
-
-  private Set<TaskSideInputHandler> getSideInputHandlers() {
-    return this.sspSideInputHandlers.values().stream().collect(Collectors.toSet());
-  }
 
   public void start() throws SamzaException, InterruptedException {
-    // Restores and recreates
+    // Restores and recreates stores.
     restoreStores();
+
     // Shutdown restore executor since it will no longer be used
     try {
       restoreExecutor.shutdown();
       if (restoreExecutor.awaitTermination(RESTORE_THREAD_POOL_SHUTDOWN_TIMEOUT_SECONDS, TimeUnit.MILLISECONDS)) {
         restoreExecutor.shutdownNow();
       }
     } catch (Exception e) {
-      LOG.error(e.getMessage());
-    }
-    if (this.hasSideInputs) {
-      startSideInputs();
+      LOG.error("Error shutting down restore executor", e);
     }
+
+    // create and restore side input stores
+    this.sideInputsManager = new SideInputsManager(
+        sideInputSystemStreams, systemFactories, activeTaskChangelogSystemStreams, storageEngineFactories, storeDirectoryPaths, containerModel, jobContext, containerContext, samzaContainerMetrics, taskInstanceMetrics, taskInstanceCollectors, streamMetadataCache, systemAdmins, serdeManager,
```

Comment:
shouldn't `activeTaskChangelogSystemStreams` be replaced by `changelogSystemStreams` passed in the constructor of `ContainerStorageManager`? 

e.g., 
```
  */
  @VisibleForTesting
  Map<TaskName, Map<String, Set<SystemStreamPartition>>> getTaskSideInputSSPs(ContainerModel containerModel,
      Map<String, Set<SystemStream>> sideInputSystemStreams, Map<String, SystemStream> changelogSystemStreams) {
```
method used in CSM prior to refactor which has the caller arguments
```this.taskSideInputStoreSSPs = getTaskSideInputSSPs(containerModel, sideInputSystemStreams, changelogSystemStreams);```

Reply:
Good catch! Fixed and verified and renamed other usages to clearly distinguish between the two.